src/App.tsx

```diff
@@ -1,24 +1,97 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { getTodos } from './api/todos';
+import { Loader } from './components/Loader';
+import { formatTodos } from './utils/formatResponse';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>();
+  const [errors, setErrors] = useState<Error[]>([{
+    title: 'test error',
+    isImportant: false,
+  }]);
+  const [todosToRender, setTodosToRender] = useState<Todo[]>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isTempLoading, setIsTempLoading] = useState(false);
+  const [toBeCleared, setToBeCleared] = useState<Todo[]>();
+
+  const pushError = (title = 'Unable to load') => {
+    setErrors((prev: Error[] | undefined) => {
```

Comment:
You don't need to provide a type for 'prev' here, as it will be inferred from the state. You can remove '(prev: Error[] | undefined)'.